internal/splunk/httprequest.go

```diff
@@ -44,12 +44,7 @@ func HandleHTTPCode(resp *http.Response) error {
 		err = exporterhelper.NewThrottleRetry(err, time.Duration(retryAfter)*time.Second)
 	// Check for permanent errors.
 	case http.StatusBadRequest, http.StatusUnauthorized:
-		dump, err2 := httputil.DumpResponse(resp, true)
-		if err2 == nil {
-			err = consumererror.NewPermanent(fmt.Errorf("%w", fmt.Errorf("%q", dump)))
-		} else {
-			err = multierr.Append(err, err2)
-		}
+		err = consumererror.NewPermanent(err)
```

Comment:
I also think this change deserves its own PR, given that this code is shared with the splunkhec exporter.

Reply:
I had to keep the change in so that on permanent errors it is also returned. 

I also have some concerns around returning the dumped HTTP response.